### ISSUE TITLE
Unhandled Exception Cast Null to String

### ISSUE THREAD

**Issue by lzardy** [izarooni/WzVisualizer#2]
### **File:**

Skill.wz (file removed)

### **Details:** 
![WzVisualizer_TjHQWWc4yN](https://user-images.githubusercontent.com/6063922/85090812-07202c00-b1b4-11ea-82e5-337c0df7e8a8.png)

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.InvalidCastException: Unable to cast object of type 'MapleLib.WzLib.WzProperties.WzNullProperty' to type 'MapleLib.WzLib.WzProperties.WzStringProperty'.
   at WzVisualizer.WzStringUtility.GetSkill(String ID)
   at WzVisualizer.MainForm.AddGridRow(DataGridView grid, Object wzObject)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at WzVisualizer.MainForm.LoadWzData(WzMapleVersion mapleVersion, String mapleDirectory)
   at WzVisualizer.MainForm.BtnWzLoad_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3801.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
WzVisualizer
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/users/Owner/My%20Documents/DESKTOP%20STUFF/Masters/Maplestory%20pub.%20%26%20priv/v.188/WzVisualizer.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3801.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
MapleLib
    Assembly Version: 2.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/users/Owner/My%20Documents/DESKTOP%20STUFF/Masters/Maplestory%20pub.%20%26%20priv/v.188/WzVisualizer.exe
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
NAudio
    Assembly Version: 1.8.5.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/users/Owner/My%20Documents/DESKTOP%20STUFF/Masters/Maplestory%20pub.%20%26%20priv/v.188/WzVisualizer.exe
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

**Comment by izarooni** [comment]
Could you pull the latest commit and try again? I can't open your Skill.wz without String.wz and I don't think I have the WZ encryption for that file.

**Comment by lzardy** [comment]
> Could you pull the latest commit and try again? I can't open your Skill.wz without String.wz and I don't think I have the WZ encryption for that file.

### Updated to Build: release v1.4.2
### File(s):
[Skill.wz & String.wz](https://drive.google.com/file/d/1eXGl5RCqpxcL-YiNEEl6BDzQ2bQ0vBuy/view?usp=sharing)

![WzVisualizer_kdLVECYcpq](https://user-images.githubusercontent.com/6063922/85187227-18c50a80-b26c-11ea-951a-3b14e85c4b02.png)

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.InvalidCastException: Unable to cast object of type 'MapleLib.WzLib.WzProperties.WzUOLProperty' to type 'MapleLib.WzLib.WzProperties.WzCanvasProperty'.
   at WzVisualizer.MainForm.AddGridRow(DataGridView grid, Object wzObject)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at WzVisualizer.MainForm.LoadWzData(WzMapleVersion mapleVersion, String mapleDirectory)
   at WzVisualizer.MainForm.BtnWzLoad_Click(Object sender, EventArgs e)
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3801.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
WzVisualizer
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/users/Owner/My%20Documents/DESKTOP%20STUFF/Masters/Maplestory%20pub.%20%26%20priv/v.188/WzVisualizer.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3801.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
MapleLib
    Assembly Version: 2.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/users/Owner/My%20Documents/DESKTOP%20STUFF/Masters/Maplestory%20pub.%20%26%20priv/v.188/WzVisualizer.exe
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
NAudio
    Assembly Version: 1.8.5.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/users/Owner/My%20Documents/DESKTOP%20STUFF/Masters/Maplestory%20pub.%20%26%20priv/v.188/WzVisualizer.exe
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

**Comment by izarooni** [comment]
The program can open the Skill.wz file now (only tested locally) although a small amount of pictures that use the outlink/inlink property are missing, while most work. I'm not sure why... 

Anyways let me know if it works, here's the exe if you need
[WzVisualizer.zip](https://github.com/izarooni/WzVisualizer/files/4807307/WzVisualizer.zip)

**Comment by lzardy** [comment]
All skills **load successfully** given the above build. All details regarding each skill however, do not all load successfully, as you said with icons. Once the latest release build is uploaded I will close this issue and open new issues (not related to missing icons).

**Comment by izarooni** [comment]
I've posted a new release which fixes the images not showing. I ended up taking a break after not being able to figure out the problem but when I finally did, I facepalmed pretty hard. The problem was a single line of my own doing 2 years ago.
If you can confirm the images load properly we can close this issue. Thanks!

**Comment by lzardy** [comment]
Latest release (1.5.0) has fixed the load issue, image issue, and search issue (although not mentioned).
Issue resolved via **#37** e24970e06832453bd8042bf7b86a7b17291b7d19.
